inc/ticket.class.php

```diff
@@ -830,4 +853,83 @@ static function filter_actors(array $params = []): array {
 
        return $params;
     }
+
+    public static function addToTimeline($options)
```

Comment:
If you are looking for groups in the tickets' entity that are not currently assigned to the ticket, I think the code could be simplified like this:
```php
$groups = (new Group())->find([
   'NOT' => [
      // Group currently assigned to the ticket
      'id' => new QuerySubQuery([
         'SELECT' => 'groups_id',
         'FROM'   => Group_Ticket::getTable(),
         'WHERE'  => [
            'tickets_id' => $options['item']->getID(),
            'type' => CommonITILActor::ASSIGN,
         ]
      ])
   ],
   // Restrict to ticket entity
   getEntitiesRestrictCriteria(
      Group::getTable(),
      '',
      $options['item']->fields['entities_id']
   )
]);
```